realtime/src/main/java/com/mx/path/service/connection/realtime/MdxRealtimeUsersConnection.java

```diff
@@ -0,0 +1,229 @@
+        .getStatus();
+  }
+
+  /**
+   * Updates a user
+   *
+   * @param userId
+   * @param user
+   * @return
+   */
+  public MdxRealtimeResponse<MdxUser> update(String userId, MdxUser user) {
+    MdxUserWrapper mdxUserWrapper = new MdxUserWrapper();
+    mdxUserWrapper.setUser(user);
+
+    MediaType mdxType = new MediaType("application", "vnd.moneydesktop.mdx.v5+json");
+
+    return connection.request("/" + connection.getConfig().getClientId() + "/users/" + userId + ".json")
+        .withContentType(mdxType.toString())
+        .withAccept(mdxType.toString())
+        .withHeaders(headers -> {
+          headers.put("Accept-Encoding", "gzip, deflate, br");
+          headers.put("MD-API-KEY", connection.getConfig().getApiKey());
+        })
+        .withBody(GSON.toJson(mdxUserWrapper))
+        .withOnComplete(response -> {
+          HttpStatus status = response.getStatus();
+          if (status == HttpStatus.UNAUTHORIZED) {
+            throw new UnauthorizedException("Encountered an authorization error updating MDX user", "Encountered an authorization error updating MDX user").withReport(true);
+          }
+        })
+        .withProcessor(response -> {
+          MdxRealtimeResponse<MdxUser> mdxRealtimeResponse = new MdxRealtimeResponse<>(response.getStatus(), null);
+          if (response.getStatus() == HttpStatus.OK) {
+            mdxRealtimeResponse.setObject(GSON.fromJson(response.getBody(), MdxUserWrapper.class).getUser());
+          }
+          return mdxRealtimeResponse;
+        })
+        .put()
+        .throwException()
+        .getObject();
+  }
+
+  /**
+   * Re-maps a user id
+   *
+   * @param userId
+   * @param newUserId
+   * @return
+   */
+  public MdxRealtimeResponse<MdxUser> remapId(String userId, String newUserId) {
+    MdxUserWrapper mdxUserWrapper = new MdxUserWrapper();
+    MdxUser user = new MdxUser();
+    user.setId(newUserId);
+    mdxUserWrapper.setUser(user);
+
+    MediaType mdxType = new MediaType("application", "vnd.moneydesktop.mdx.v5+json");
+
+    return connection.request("/" + connection.getConfig().getClientId() + "/users/" + userId + "/remap_id.json")
+        .withContentType(mdxType.toString())
+        .withAccept(mdxType.toString())
+        .withHeaders(headers -> {
+          headers.put("Accept-Encoding", "gzip, deflate, br");
+          headers.put("MD-API-KEY", connection.getConfig().getApiKey());
+        })
+        .withBody(GSON.toJson(mdxUserWrapper))
+        .withOnComplete(response -> {
+          HttpStatus status = response.getStatus();
+          if (status == HttpStatus.UNAUTHORIZED) {
+            throw new UnauthorizedException("Encountered an authorization error remapping MDX user", "Encountered an authorization error remapping MDX user").withReport(true);
+          }
+        })
+        .withProcessor(response -> {
+          MdxRealtimeResponse<MdxUser> mdxRealtimeResponse = new MdxRealtimeResponse<>(response.getStatus(), null);
```

Comment:
It looks like this constructor is always passed a null for the object parameter. Maybe we can remove that from the constructor params.

Reply:
Fixed